x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/enrollment/tool/BootstrapPasswordAndEnrollmentForInitialNode.java

```diff
@@ -0,0 +1,196 @@
+        return clientFunction.apply(env);
+    }
+    CreateEnrollmentToken getCreateEnrollmentToken(Environment env) throws Exception{
+        return createEnrollmentTokenFunction.apply(env);
+    }
+    KeyStoreWrapper getKeyStoreWrapper(Environment env) throws Exception { return keyStoreFunction.apply(env); }
+    OptionParser getParser() { return parser; }
+
+    BootstrapPasswordAndEnrollmentForInitialNode() {
+        this(
+            environment -> new CommandLineHttpClient(environment),
+            environment -> KeyStoreWrapper.load(environment.configFile()),
+            environment -> new CreateEnrollmentToken(environment)
+        );
+    }
+
+    BootstrapPasswordAndEnrollmentForInitialNode(Function<Environment, CommandLineHttpClient> clientFunction,
+                                                 CheckedFunction<Environment, KeyStoreWrapper, Exception> keyStoreFunction,
+                                                 CheckedFunction<Environment, CreateEnrollmentToken, Exception>
+                                                     createEnrollmentTokenFunction){
+        super(clientFunction, keyStoreFunction, "Set elastic password and generate enrollment token for initial node");
+        this.createEnrollmentTokenFunction = createEnrollmentTokenFunction;
+        parser.allowsUnrecognizedOptions();
+    }
+
+    public static void main(String[] args) throws Exception {
+        exit(new BootstrapPasswordAndEnrollmentForInitialNode().main(args, Terminal.DEFAULT));
+    }
+
+    @Override
+    protected void execute(Terminal terminal, OptionSet options, Environment env) throws Exception {
+        if (options.nonOptionArguments().contains("--explicitly-acknowledge-execution") == false) {
+            throw new UserException(ExitCodes.CONFIG, "This command is not intended for end users");
+        }
+        if (env.settings().hasValue(XPackSettings.ENROLLMENT_ENABLED.getKey()) && false ==
+            XPackSettings.ENROLLMENT_ENABLED.get(env.settings())) {
+            throw new UserException(ExitCodes.NOOP, "Enrollment is explicitly disabled.");
+        }
+        final CommandLineHttpClient client = getClient(env);
+        final CreateEnrollmentToken createEnrollmentToken = getCreateEnrollmentToken(env);
+        keyStoreWrapper = getKeyStoreWrapper(env);
+        ReadBootstrapPassword(env, terminal);
+        checkClusterHealthWithRetries(client, terminal, elasticUser, credentialsPassword, 0, 5, false);
+        if (Strings.isNullOrEmpty(bootstrapPassword.toString())) {
+            changeElasticUserPassword(terminal, client);
+        } else {
+            password = bootstrapPassword;
+        }
+        token = createEnrollmentToken.createKibanaEnrollmentToken(elasticUser, password);
+        fingerprint = createEnrollmentToken.getFingerprint();
+
+        if (Strings.isNullOrEmpty(bootstrapPassword.toString())) {
+            terminal.println("'elastic' user password: " + password);
```

Comment:
These will need to be parsed by another process and not seen directly from users so we can make them machine friendly ( i.e. skip single quotes)